servers/plugins/org.fusesource.ide.server.karaf.core/src/org/fusesource/ide/server/karaf/core/publish/jmx/KarafBundleMBeanPublishBehaviour.java

```diff
@@ -159,4 +160,14 @@ public boolean canHandle(MBeanServerConnection mbsc) {
 
 	    return false;
 	}
+	
+	protected String getBundleSymbolicName(CompositeData cd) {
+		String bsn = null;
+		try {
+			bsn = cd.get("Symbolic Name").toString();
```

Comment:
isn't the differecne between Karaf 2-3 and 4? if yes, what do you thinkto have a specific class for each case? or at least to extract methods mentioning that one is for 4- and the other for 4+?

Reply:
the point is I don't know if thats true. Could be also that latest 2.4 or 3 versions changed that too.

Reply:
And its just this one logic change so far. If we get more then it makes probably sense.